src/transaction/lock.rs

```diff
@@ -132,6 +142,277 @@ async fn resolve_lock_with_retry(
     Err(error.expect("no error is impossible"))
 }
 
+#[derive(Default, Clone)]
+pub struct ResolveLocksContext {
+    // Record the status of each transaction.
+    pub(crate) resolved: Arc<RwLock<HashMap<u64, Arc<TransactionStatus>>>>,
+    pub(crate) clean_regions: Arc<RwLock<HashMap<u64, HashSet<RegionVerId>>>>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct ResolveLocksOptions {
+    pub async_commit_only: bool,
```

Comment:
Why providing this option?

Reply:
It is used for cloud-engine native restore.
We only need to resolve async commit locks, other locks can be resolved later by tidb.